src/ui_nbgl.c

```diff
@@ -81,41 +92,72 @@ static bool navigation_cb_baking(uint8_t page, nbgl_pageContent_t* content) {
 
     TZ_ASSERT(hwm_to_string(buffer[2], sizeof(buffer[2]), &N_data.hwm.main) >= 0, EXC_WRONG_LENGTH);
 
-    if (page == 0u) {
-        content->type = INFOS_LIST;
-        content->infosList.nbInfos = 3;
-        content->infosList.infoTypes = bakeInfoTypes;
-        content->infosList.infoContents = bakeInfoContents;
-    } else {
-        content->type = INFOS_LIST;
-        content->infosList.nbInfos = 3;
-        content->infosList.infoTypes = infoTypes;
-        content->infosList.infoContents = infoContents;
+    switch (page) {
+        case 0:
+            content->type = INFOS_LIST;
+            content->infosList.nbInfos = 3;
+            content->infosList.infoTypes = bakeInfoTypes;
+            content->infosList.infoContents = bakeInfoContents;
+            break;
+        case 1:
+            content->type = INFOS_LIST;
+            content->infosList.nbInfos = 3;
+            content->infosList.infoTypes = infoTypes;
+            content->infosList.infoContents = infoContents;
+            break;
+        case 2:
+            switches[HWM_ENABLED_TOKEN_ID].initState = (nbgl_state_t) (!hwm_disabled);
+            switches[HWM_ENABLED_TOKEN_ID].text = "High Watermark";
+            switches[HWM_ENABLED_TOKEN_ID].subText = "Track high watermark\n in Ledger";
+            switches[HWM_ENABLED_TOKEN_ID].token = HWM_ENABLED_TOKEN;
+            switches[HWM_ENABLED_TOKEN_ID].tuneId = TUNE_TAP_CASUAL;
+            content->type = SWITCHES_LIST;
+            content->switchesList.nbSwitches = SETTINGS_SWITCHES_NB;
+            content->switchesList.switches = (nbgl_layoutSwitch_t*) switches;
+            break;
```

Comment:
Personal opinion:
I think that the Copyright/version should remain at the end and that the parameters should be more accessible.
`Baking data > Settings > Copyright`

Reply:
Done.